content/2022/prepoznavanje-govora.md

```diff
@@ -2,3 +2,239 @@
+
+ReLU (rectified linear activation function / rectified linear unit) je funkcija koja negativnim vrednostima daje nulu, a pozitivne ostavlja kakve jesu. Time dobijamo nelinearan model.
+
+![Funkcija](static\images\fja.png)
+
+Kroz neuronsku mrežu se propušta već napravljen spektrogram, kao i labele tih spektrograma koje mreža treba da prepozna.
+
+Za treniranje mreže koriste se dve metode simultano (propagacija unapred i unazad), kao i jedna funkcija (kriterijumska funkcija)
+
+Kritetijumska funkcija (eng. cost funkcija) jeste funkcionalna veza željenog outputa i dobijenog outputa u funkciji. Takođe, kriterijumska funkcija je usrednjena vrednost svih funkcija greške.
+
+Najkorišćenija loss funkcija je Cross Entropy Loss. Potrebno nam je da minimizujemo grešku unakrsne entropije za što preciznije rezultate.
+
+Formula po kojoj se računa Cross Entropy Loss je sledeća:
+
+$$L_{C E}(\hat{y}, y)=-[y \log \sigma(\mathbf{w} \cdot \mathbf{x}+b)+(1-y) \log (1-\sigma(\mathbf{w} \cdot \mathbf{x}+b))]$$
+
+Propagacija unazad je metod smanjenja grešaka u CNN posmatranjem neophodnih promena vrednosti parametara mreže u svakom sloju kako bi se neuroni aktivirali na određen način.
+
+Propagacija unazad prolazi kroz sve slojeve i menja parametre mreže u svakom koraku.
+
+Parametri mreže se menjaju u cilju računanja dovoljno dobrog gradient spusta za traženje lokalnog / maksimalnog minimuma ove funkcije. Dakle, teži se tome da gradijent kriterijumske funkcije bude što bliži nuli.
+
+![SGD](static/images/sgd.png)
+
+### Istraživanje i rezultati
+
+Testiranje metoda vršeno je na dve baze: FSDD i srpske baze kreirane za potrebe projekta.
+
+FSDD baza sadrži engleske cifre od 0 do 9 koje su izgovorene od strane 50 različitih ljudi. Sadrži ukupno 3000 snimaka.
+
+Srpska baza sadrži 10 srpskih reči, gde su specifično birane reči koje su slične po nekim karakteristikama (ponavljanje slova, zamena slova, umanjenice, ...). Baza ukupno sadrži 500 snimaka, gde preko 10 ljudi izgovara ove reči različitim naglaskom, intonacijom i slično.
+
+Za konvolucionu neuronsku mrežu, potrebni su nam bili pokazatelji kako mreža uči tokom epoha treniranja. Baze su podeljene na trening, test i validacionu bazu, tako da je trening set sadržao 70% reči, a test i validacioni set po 15% reči u slučaju obe baze.
+
+Rezultati su prikazani u tabeli ispod.
+
+![Rezultati](static\images\4.png)
+
+Metrika ovih rezultata bila je tačnost, zato što je, zbog izbalansirane baza, ovo reprezentativna metrika.
+
+Rezultati se dele po tome da li metoda koristi duboko učenje ili ne. Posmatrajući tabelu, konvoluciona neuronska mreža je ostvarila najveću tačnost kao metoda sa dubokim učenjem, a XGBoost daje najbolje rezultate među metodama koje ne koriste duboko učenje. Najmanje rezultate daje SVM sa polinomijalnim kernelom.
+
+U FSDD bazi dato je 10 labela, pa klasifikator radi odličan posao da pretpostavi u koju kategoriju labela određeni zvuk spada (cifra od 0 do 9).
+
```

Comment:
Okej, ovde nam treba malo duzi deo o matricama konfuzije. Evo kako to treba da izgleda:
Prikazane su matrice konfuzije na FSDD bazi za CNN, XGBoost, SVM, Random Forest i Logisticku regresiju. 
* slike matrica
Iz ovih matrica konfuzije, mozemo da zakljucimo sledece:
(Ovde vi treba da gledate matrice i pokusate da vidite neke pravilnosti ili zanimljivosti.
Evo jedna pomoc za pocetak: Primetite da je prisutan sablon da se zvuk jedinice cesto pogresno klasifikuje kao zvuk devetke. Diskutujte da li to ima nekog fizickog smisla, iz pogleda nacina izgovora "one" i "nine")